api/envoy/api/v2/core/base.proto

```diff
@@ -150,6 +150,15 @@ message RuntimeUInt32 {
   string runtime_key = 3 [(validate.rules).string = {min_bytes: 1}];
 }
 
+// Runtime derived bool with a default when not specified.
+message RuntimeFeatureFlag {
+  // Default value if runtime value is not available.
+  bool default_value = 1;
+
+  // Runtime key to get value for comparison. This value is used if defined.
```

Comment:
Do you want % based rollout here? Or do you want a hard true/false? I think either is fine but it's not completely clear to me what should be in runtime here. What values count as true and false?

Reply:
> Do you want % based rollout here?


I think this should be all or nothing. Percentage based rollout wouldn't make much sense since the requests that get to bypass the filter logic will still have impact on the upstream. That would cause some strange behavior in the filter's feedback loop.

> What values count as true and false?


It's a scalar, so it would be represented as a JSON bool, no? I was going off of what's mentioned in the docs and from other examples of bools in our .proto files:
https://www.envoyproxy.io/docs/envoy/latest/configuration/operations/runtime.html#protobuf-and-json-representation

Reply:
I'm not sure what the runtime text file should be. I.e., true/false or 0/1 or what? Can you just link back to bool documentation for runtime and make some if there isn't any?